core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/neo/PauseUtil.scala

```diff
@@ -0,0 +1,74 @@
+package edu.uci.ics.amber.engine.architecture.worker.neo
+
+import java.util.concurrent.CompletableFuture
+
+object PauseUtil{
+  // TODO: check if this is necessary
```

Comment:
we need to discuss this separately, it will be good if there are some examples on how this can benefit the code. 

I don't like this design, making each functionality having a privilege level feels too random/arbitrary. for example, "backpressure" < "breakpoint", we don't even implement backpressure and how can we know it's less than breakpoint? and "coreException" < "internal" < "user", and what's the exact definition of each? it's not good to "forward" thinking too much, since the future requirements are unclear

I have a feeling that this will be another source of many unexpected behaviors, bugs, and frustrations. like "why it's not paused when it should pause" and the reason is the pause level is somehow messed up, or there might be two  different cases that we want the pause precedence to be different....

I suggest we discuss based on various concrete examples, like "breakpoint pauses -> before UI changed, user pauses", or "another actor pauses -> before the message is sent, this actor also pauses",  we need to list as many examples like this as possible to discuss different behaviors of pause/resume under different cases

Reply:
I agree with @zuozhiw . The scenario @shengquan-ni has thought is valid, but we need more discussion.